core/templates/dev/head/pages/exploration_editor/editor_tab/SolutionValidityService.js

```diff
@@ -0,0 +1,24 @@
+// Service for keeping track of solution validity.
+oppia.factory('SolutionValidityService', [
```

Comment:
Yes, I did spend a lot of time on that. Just to check -- I should be creating a solution, verifying it and setting a flag and then checking if the flag is set/unset, right? I mean these are the kind of tests that should be here?